File: November 2022/901. Online Stock Span/solution.java
// Asked on 09-11-2022 (DD-MM-YYYY)
// Language: Java

class StockSpanner {
  Stack < int[] > s;
  public StockSpanner() {
    s = new Stack < > ();
  }
  public int next(int price) {
    int span = 1;
    while (!s.isEmpty() && price >= s.peek()[0]) {
      span += s.peek()[1];
      s.pop();
    }
    s.push(new int[] {
      price,
      span
    });
    return span;
  }
}

File: November 2022/901. Online Stock Span/solution.py
# Asked on 09-11-2022 (DD-MM-YYYY)
# Language: Python

class StockSpanner:
    def __init__(self):
        self.stack = [[inf, 0]]

    def next(self, price: int) -> int:
        curDay = self.stack[-1][1] + 1

        while price >= self.stack[-1][0]:
            self.stack.pop(-1)

        span = curDay - self.stack[-1][1]
        self.stack.append([price, curDay])

        return span
